lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1336,6 +1349,29 @@ bool WitnessChecker::findBestWitness(
       auto match = matchWitness(ReqEnvironmentCache, Proto, conformance,
                                 DC, requirement, witness);
       if (match.isViable()) {
+        if (canDerive && isSwiftRawRepresentableEnum &&
+            isEquatableConformance) {
+          // For enum types that can derive equatable conformance, let's always
```

Comment:
This seems like an overly specific check to add to findBestWitness(). We wouldn't want multiple bits of code like this to accumulate here over time. I would prefer it if we could find another solution here.

Reply:
Thanks for the review @slavapestov! 

>  We wouldn't want multiple bits of code like this to accumulate here over time. I would prefer it if we could find another solution here.

That makes sense, I'll try to come up with a different approach :)

Reply:
Made some adjustments in the approach, let me know what you think